core/sql/generator/GenPreCode.cpp

```diff
@@ -10715,7 +10808,10 @@ RelExpr * PhysicalFastExtract::preCodeGen (Generator * generator,
   if (nodeIsPreCodeGenned())
     return this;
 
-  generator->setIsFastExtract(TRUE);  
+  if (getIsMainQueryOperator())
+    generator->setIsFastExtract(TRUE);
+  else
+    generator->setContainsFastExtract(TRUE);
```

Comment:
The "is" and the "contains" refer to the entire statement. The Generator::isFastExtract() method is used to determine the overall statement type. What I needed was a flag indicating that there is some fast extract somewhere in the statement (e.g. the one used to populate a temp table), while the overall statement type is something else, like a select. I added a comment to make that more clear.